bench: refactor random number generation in `stats/base/dists/uniform`

MunishMummadi opened the pull request:
This issue is related to #4991 i.e RFC:

## Description

This PR has changes to benchmarking file located at [stats/base/dists/uniform](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/uniform)

This pull request:

-   Replaced `rand()` with corresponding `uniform()` implementation within the benchmarking files.



## Questions

No

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -38,9 +38,9 @@ bench( pkg, function benchmark( b ) {
-		max = min + ( randu() * 40.0 );
+		x = uniform( -10.0, 10.0);
+		min = uniform( -20.0, 0.0);
+		max = min + uniform( 0.0, 40.0);
```
The random number generation should be moved outside the benchmarking loop, before `b.tic()`.

MunishMummadi replied:
> The random number generation should be moved outside the benchmarking loop, before `b.tic()`.

I have a question about this suggestion. Are you expecting this change only for this file or to the complete PR.

anandkaranubc replied:
Yes, this change is expected in the complete PR. One of the main reasons for this refactoring is to ensure that random number generation does not interfere with benchmarking. Reference PRs would be a good source to verify if your changes are correct.

MunishMummadi replied:
Thanks for the confirmation. I thought so, I will update soon.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/ctor/benchmark/benchmark.js:
```diff
@@ -92,7 +92,7 @@ bench( pkg+'::set:a', function benchmark( bm ) {
 	bm.tic();
 	for ( i = 0; i < bm.iterations; i++ ) {
-		y = ( 100.0*randu() ) + EPS;
+		y = uniform( EPS, 100.0);
```
```suggestion
		y = uniform( EPS, 100.0 );
```

This change should be applied consistently throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/ctor/benchmark/benchmark.js:
```diff
@@ -145,7 +145,7 @@ bench( pkg+'::set:b', function benchmark( bm ) {
 	bm.tic();
 	for ( i = 0; i < bm.iterations; i++ ) {
-		y = ( 100.0*randu() ) + a + EPS;
+		y = uniform(a + EPS, 100.0);
```
```suggestion
		y = uniform( a + EPS, 100.0 );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/entropy/benchmark/benchmark.js:
```diff
@@ -41,8 +41,8 @@ bench( pkg, function benchmark( b ) {
-		min[ i ] = ( randu()*10.0 );
-		max[ i ] = ( randu()*10.0 ) + min[ i ];
+		min[ i ] = uniform( 0.0, 10.0 );
+		max[ i ] = uniform( 0.0, 10.0 ) + min[ i ];
```
```suggestion
		max[ i ] = uniform( min[ i ], min[ i ] + 10.0 );
```

anandkaranubc commented:
Thanks, @MunishMummadi, for working on this! I've left a few suggestions for you to review.

Also, can you change the title to:

bench: refactor random number generation in `stats/base/dists/uniform`

MunishMummadi commented:
Sure, I have updated the title.

MunishMummadi commented:
@anandkaranubc I don't see any code review. If that's what you mean by suggestions. Please correct me if I am wrong.

anandkaranubc commented:
> @anandkaranubc I don't see any code review. If that's what you mean by suggestions. Please correct me if I am wrong.

Sorry, my bad. They should be visible now.

MunishMummadi committed:
feat: updated

MunishMummadi commented:
@anandkaranubc I also added changes where they seem fit. i have doubts about the `b.tic()` i commented about it on the code review.
edit: changed to changes

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/uniform/cdf/benchmark/benchmark.js:
```diff
@@ -36,11 +36,12 @@ bench( pkg, function benchmark( b ) {
 	var y;
 	var i;
 
+	x = uniform( -10.0, 10.0 );
+	min = uniform( -20.0, 0.0 );
+	max = uniform( min, min + 40.0 );
+
```
```suggestion
	len = 100;
	x = new Float64Array( len );
	min = new Float64Array( len );
	max = new Float64Array( len );
	for ( i = 0; i < len; i++ ) {
		x[ i ] = uniform( -10.0, 10.0 );
		min[ i ] = uniform( -20.0, 0.0 );
		max[ i ] = uniform( min[ i ], min[ i ] + 40.0 );
	}

```

This would be the correct way to initialize random values for the variables.

MunishMummadi replied:
Thanks for this update. This really helps!!!

Planeshifter commented:
Meanwhile https://github.com/stdlib-js/stdlib/pull/5176 was merged, so this PR won't be needed anymore. Thanks for your efforts, @MunishMummadi, which we appreciate!
